backend/internal/api/routes.go

```diff
@@ -20,6 +22,36 @@ func SetupRoutes(router *gin.Engine) {
 	}
 }
 
+// SetupRoutesWithRepos configures all API routes with repository access
+func SetupRoutesWithRepos(router *gin.Engine, repos *repository.Repositories, cfg *config.Config) {
+	router.Use(CORSMiddleware())
+
+	// Create handler with repositories
+	h := &Handler{
+		repos: repos,
+		cfg:   cfg,
+	}
+
+	api := router.Group("/api/v1")
+	{
+		api.GET("/health", healthCheck)
+
+		// Graph endpoints
+		api.GET("/graph", h.getGraph)
+		api.GET("/graph/viewport", h.getViewportNodes)
+		
+		// Node endpoints
+		api.GET("/nodes/:id", h.getNode)
+		api.GET("/nodes/:id/content", h.getNodeContent)
+		api.PUT("/nodes/:id/position", h.updateNodePosition)
+		api.GET("/nodes/search", h.searchNodes)
+		
+		// Vault management endpoints (for future use)
+		api.POST("/vault/parse", h.parseVault)
+		api.GET("/vault/status", h.getParseStatus)
+	}
+}
+
```

Comment:
Great addition of new endpoints! The search and vault management endpoints (parseVault, getParseStatus) align well with the roadmap. The parseVault endpoint correctly returns 501 Not Implemented, clearly indicating future work.